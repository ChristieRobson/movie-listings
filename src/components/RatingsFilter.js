import React from 'react';
import { Label, Input, InputGroup } from 'reactstrap';

export const RatingsFilter = (props) => {
    return (
        <div className="row">
            <div className="col-12 col-md-6 col-lg-4">
                <InputGroup>
                    <Label for="min-rating">Minimum rating</Label>
                    <Input type="number"
                           id="min-rating"
                           min="0"
                           max="10"
                           step="0.5"
                           value={props.value}
                           onChange={props.onChange}
                    />
                </InputGroup>
            </div>
        </div>
    )
};
