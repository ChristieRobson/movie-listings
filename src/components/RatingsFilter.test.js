import React from 'react';
import { shallow } from 'enzyme';
import { RatingsFilter } from './RatingsFilter';

it('renders RatingsFilter', () => {
    const value = 5;
    const noop = () => {};
    let component = shallow(<RatingsFilter value={value} onChange={noop}/>);
    expect(component).toMatchSnapshot();
});