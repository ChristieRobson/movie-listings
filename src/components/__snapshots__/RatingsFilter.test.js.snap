// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RatingsFilter 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    onChange={[Function]}
    value={5}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="col-12 col-md-6 col-lg-4"
>
        <InputGroup
                tag="div"
        >
                <Label
                        for="min-rating"
                        tag="label"
                        widths={
                                Array [
                                        "xs",
                                        "sm",
                                        "md",
                                        "lg",
                                        "xl",
                                      ]
                        }
                >
                        Minimum rating
                </Label>
                <Input
                        id="min-rating"
                        max="10"
                        min="0"
                        onChange={[Function]}
                        step="0.5"
                        type="number"
                        value={5}
                />
        </InputGroup>
</div>,
      "className": "row",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <InputGroup
          tag="div"
>
          <Label
                    for="min-rating"
                    tag="label"
                    widths={
                              Array [
                                        "xs",
                                        "sm",
                                        "md",
                                        "lg",
                                        "xl",
                                      ]
                    }
          >
                    Minimum rating
          </Label>
          <Input
                    id="min-rating"
                    max="10"
                    min="0"
                    onChange={[Function]}
                    step="0.5"
                    type="number"
                    value={5}
          />
</InputGroup>,
        "className": "col-12 col-md-6 col-lg-4",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Label
              for="min-rating"
              tag="label"
              widths={
                            Array [
                                          "xs",
                                          "sm",
                                          "md",
                                          "lg",
                                          "xl",
                                        ]
              }
>
              Minimum rating
</Label>,
            <Input
              id="min-rating"
              max="10"
              min="0"
              onChange={[Function]}
              step="0.5"
              type="number"
              value={5}
/>,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Minimum rating",
              "for": "min-rating",
              "tag": "label",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": "Minimum rating",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "min-rating",
              "max": "10",
              "min": "0",
              "onChange": [Function],
              "step": "0.5",
              "type": "number",
              "value": 5,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="col-12 col-md-6 col-lg-4"
>
          <InputGroup
                    tag="div"
          >
                    <Label
                              for="min-rating"
                              tag="label"
                              widths={
                                        Array [
                                                  "xs",
                                                  "sm",
                                                  "md",
                                                  "lg",
                                                  "xl",
                                                ]
                              }
                    >
                              Minimum rating
                    </Label>
                    <Input
                              id="min-rating"
                              max="10"
                              min="0"
                              onChange={[Function]}
                              step="0.5"
                              type="number"
                              value={5}
                    />
          </InputGroup>
</div>,
        "className": "row",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <InputGroup
            tag="div"
>
            <Label
                        for="min-rating"
                        tag="label"
                        widths={
                                    Array [
                                                "xs",
                                                "sm",
                                                "md",
                                                "lg",
                                                "xl",
                                              ]
                        }
            >
                        Minimum rating
            </Label>
            <Input
                        id="min-rating"
                        max="10"
                        min="0"
                        onChange={[Function]}
                        step="0.5"
                        type="number"
                        value={5}
            />
</InputGroup>,
          "className": "col-12 col-md-6 col-lg-4",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Label
                for="min-rating"
                tag="label"
                widths={
                                Array [
                                                "xs",
                                                "sm",
                                                "md",
                                                "lg",
                                                "xl",
                                              ]
                }
>
                Minimum rating
</Label>,
              <Input
                id="min-rating"
                max="10"
                min="0"
                onChange={[Function]}
                step="0.5"
                type="number"
                value={5}
/>,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Minimum rating",
                "for": "min-rating",
                "tag": "label",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": "Minimum rating",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "min-rating",
                "max": "10",
                "min": "0",
                "onChange": [Function],
                "step": "0.5",
                "type": "number",
                "value": 5,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
